tensor_genn/algorithms/relu_ann.py

```diff
@@ -69,37 +117,46 @@ def convert(self, tf_model):
         
         cs_init = {"magnitude":10.0}
 
+        # Fetch augmented weight matrices
+        g_weights, n_units = self.create_weight_matrices(tf_model)
+        gw_inds = [np.nonzero(gw) for gw in g_weights]
+        gw_vals = [g_weights[i][gw_inds[i]].reshape(-1) for i in range(len(g_weights))]
+
         # Define model and populations
         self.g_model = genn_model.GeNNModel("float","g_model")
         self.neuron_pops = []
+        self.syn_pops = []
 
-        for i in range(1,n_layers): # 1,2 - synapses
+        for i in range(1,n_layers):
             if i == 1:
                 # Presynaptic neuron
                 self.neuron_pops.append(self.g_model.add_neuron_population(
-                    "if"+str(i-1),tf_weights[i-1].shape[0],if_model,if_params,if_init)
+                    "if"+str(i-1),n_units[i-1],if_model,if_params,if_init)
                 )
-
+            
             # Postsynaptic neuron
             self.neuron_pops.append(self.g_model.add_neuron_population(
-                "if"+str(i),tf_weights[i-1].shape[1],if_model,if_params,if_init)
+                "if"+str(i),n_units[i],if_model,if_params,if_init)
             )
 
             # Synapse
-            self.g_model.add_synapse_population(
-                "syn"+str(i-1)+str(i),"DENSE_INDIVIDUALG",genn_wrapper.NO_DELAY,
-                self.neuron_pops[i-1], self.neuron_pops[i],
-                "StaticPulse",{},{'g':tf_weights[i-1].reshape(-1)},{},{},
-                "DeltaCurr",{},{}
+            self.syn_pops.append(self.g_model.add_synapse_population(
```

Comment:
It's very inefficient to represent dense layers using sparse connectivity - it doubles the memory usage and memory bandwidth required for spike processing so this needs a bit more thought.